ykcompile/src/lib.rs

```diff
@@ -437,13 +510,13 @@ impl<TT> TraceCompiler<TT> {
                 // This Local lives now on the stack...
                 self.variable_location_map.insert(p2.local, loc.clone());
                 // ...so we can free its old register.
-                self.register_content_map.insert(reg, None);
+                self.register_content_map.insert(reg, RegisterVal::Free);
```

Comment:
Can this not go through the freeing method I saw before?

Reply:
Not quite. `free_register()` also sets the `variable_location_map` to `NotLive`, which we don't want since the local now lives on the stack.